cbmc/proofs/IotMqtt_Wait/IotMqtt_Wait_harness.c

```diff
@@ -0,0 +1,34 @@
+#include "iot_config.h"
+#include "private/iot_mqtt_internal.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "mqtt_state.h"
+
+void harness()
+{
+  IotMqttConnection_t mqttConnection = allocate_IotMqttConnection(NULL);
+  __CPROVER_assume(mqttConnection != NULL);
+  __CPROVER_assume(mqttConnection->pNetworkInterface != NULL);
+  __CPROVER_assume( IS_STUBBED_NETWORKIF_DESTROY( mqttConnection->pNetworkInterface ) );
+  ensure_IotMqttConnection_has_lists(mqttConnection);
+  __CPROVER_assume(valid_IotMqttConnection(mqttConnection));
+
+  IotMqttOperation_t publishOperation = allocate_IotMqttOperation(NULL, mqttConnection);
+  __CPROVER_assume(valid_IotMqttOperation(publishOperation));
+
+  if (nondet_bool())
```

Comment:
The latest version of mqtt_state now a method to allocate a list of length 0 or 1 (or 2 or 3).  This might be a good place to use this allocate_ and the corresponding valid_

Reply:
CBMC reports a `out of memory` during the generation of the `coverage.xml`. I'll keep the proof harness as it is, until we fix this in CBMC.